scripts/vippy/common.py

```diff
@@ -0,0 +1,266 @@
+# Copyright (c) 2015, Christopher Jerdonek
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the names of the copyright holders nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import logging
+import os.path
+import pprint
+
+import yaml
+
+
+_log = logging.getLogger()
+
+DATA_DIR = 'docs/data'
+TABLES_DIR = 'docs/tables'
+XML_DIR = 'docs/xml'
+
+_TAG_KEY_ON_ERROR = 'on_error'
+_TAG_KEY_ON_ERROR_CUSTOM = 'on_error_custom'
+
+TAG_KEY_NAME = '_name'
+TAG_KEY_TYPE = 'type'
+TAG_KEY_REQUIRED = 'required'
+TAG_KEY_REPEATING = 'repeating'
+TAG_KEY_DESCRIPTION = 'description'
+TAG_KEY_ERROR_HANDLE = 'error'
+
+DEFAULT_TAG_VALUES = {
+    TAG_KEY_REQUIRED: False,
+    TAG_KEY_REPEATING: False,
+}
+
+ENUMERATIONS = set([
+    'BallotMeasureType',
```

Comment:
Agreed.  For now, I can at least update the PR to get the list from the YAML files themselves (which it is already reading).  The two simple types I'll keep hard-coded though since those don't have YAML files.